initfiles/sbin/hpcc-run.sh.in

```diff
@@ -52,134 +52,143 @@ getIPS(){
 }
 
 getDali(){
-        DIP=`${INSTALL_DIR}/sbin/configgen -env ${envfile} -listall | grep Dali | awk -F, '{print \$3}'  | sort | uniq`
+    DIP=`${INSTALL_DIR}/sbin/configgen -env ${envfile} -listall | grep Dali | awk -F, '{print \$3}'  | sort | uniq`
 }
 
 createIPListFile(){
-    _file=$1
-    echo "$IPS" > $_file
+    local _output=$1
+    echo "$IPS" > $_output
 }
 
 createIPListFileExcludeDIP(){
-  _file=$1
-  echo "$IPS" | grep -v $DIP  > $_file
+    local _input=$1
+    local _output=$2
```

Comment:
If $DIP was in the form ip|ip|ip you could use grep in a single pass (might need -E option). If you do multiple passes it's not much better (probably worse) than the n^2 version we had before.

But there's another problem using grep, which is that unless you transform the ip's into appropriate patterns all the .'s in there are going to match any character.

Perhaps the version you have with the nested loops is the best after all.